pandas-stubs/io/sql.pyi

```diff
@@ -1,191 +1,139 @@
+    *,
+    chunksize: int,
+    dtype: DtypeArg | None = ...,
+) -> Iterator[DataFrame]: ...
+@overload
 def read_sql_query(
-    sql,
-    con,
-    schema: str | None = ...,
-    index_col: str | Sequence[str] | None = ...,
+    sql: str,
+    con: Any,
+    index_col: str | list[str] | None = ...,
     coerce_float: bool = ...,
-    params=...,
-    parse_dates: Sequence[str] | Mapping[str, str] | None = ...,
-    chunksize: int | None = ...,
+    params: list[str] | dict[str, str] | None = ...,
+    parse_dates: list[str] | dict[str, str] | dict[str, dict[str, Any]] | None = ...,
+    chunksize: None = ...,
+    dtype: DtypeArg | None = ...,
 ) -> DataFrame: ...
+@overload
 def read_sql(
-    sql: str | Any,
-    con: str | Any,
-    index_col: str | Sequence[str] | None = ...,
+    sql: str,
+    con: Any,
+    index_col: str | list[str] | None = ...,
     coerce_float: bool = ...,
-    params: Sequence[str] | tuple[str, ...] | Mapping[str, str] | None = ...,
-    parse_dates: Sequence[str]
-    | Mapping[str, str]
-    | Mapping[str, Mapping[str, Any]]
-    | None = ...,
-    columns: Sequence[str] = ...,
-    chunksize: int = ...,
+    params: list[str] | dict[str, str] | None = ...,
+    parse_dates: list[str] | dict[str, str] | dict[str, dict[str, Any]] | None = ...,
+    columns: list[str] = ...,
+    *,
+    chunksize: int,
+) -> Iterator[DataFrame]: ...
+@overload
+def read_sql(
+    sql: str,
+    con: Any,
+    index_col: str | list[str] | None = ...,
+    coerce_float: bool = ...,
+    params: list[str] | dict[str, str] | None = ...,
+    parse_dates: list[str] | dict[str, str] | dict[str, dict[str, Any]] | None = ...,
+    columns: list[str] = ...,
+    chunksize: None = ...,
 ) -> DataFrame: ...
-def to_sql(
-    frame,
-    name,
-    con,
-    schema=...,
-    if_exists: str = ...,
-    index: bool = ...,
-    index_label=...,
-    chunksize=...,
-    dtype=...,
-    method=...,
-) -> None: ...
-def has_table(table_name, con, schema=...): ...
-
-table_exists = has_table
-
-def pandasSQL_builder(con, schema=..., meta=..., is_cursor: bool = ...): ...
-
-class SQLTable(PandasObject):
-    name = ...
-    pd_sql = ...
-    prefix = ...
-    frame = ...
-    index = ...
-    schema = ...
-    if_exists = ...
-    keys = ...
-    dtype = ...
-    table = ...
-    def __init__(
-        self,
-        name,
-        pandas_sql_engine,
-        frame=...,
-        index: bool = ...,
-        if_exists: str = ...,
-        prefix: str = ...,
-        index_label=...,
-        schema=...,
-        keys=...,
-        dtype=...,
-    ) -> None: ...
-    def exists(self): ...
-    def sql_schema(self): ...
-    def create(self) -> None: ...
-    def insert_data(self): ...
-    def insert(self, chunksize=..., method=...) -> None: ...
-    def read(
-        self, coerce_float: bool = ..., parse_dates=..., columns=..., chunksize=...
-    ): ...
 
 class PandasSQL(PandasObject):
```

Comment:
not in docs, so delete

Reply:
I think the same comment below also applies here since this is needed to type `SQLTable`, which is then used in the `Callable` version.